README.md

```diff
@@ -1,193 +1,556 @@
+
+```bash
+$ mkdir pulumi-kotlin-demo && cd pulumi-kotlin-demo
+$ pulumi new aws-java
+```
+
+#### 3. Add GitHub Packages Maven repository.
+
+> ℹ️ GitHub Packages Maven registry requires
+> authentication, [even for public repos](https://github.com/orgs/community/discussions/25979). We will eventually get
+> rid
+> of this uneccessary step by storing all the artifacts in Maven Central.
+
+Generate [GitHub personal access token](https://github.com/settings/tokens/new) with `read:packages` scope and
+include `server` in your `~/.m2/settings.xml` file:
+
+```xml
+
+<servers>
+    <server>
+        <id>github-pulumi-kotlin</id>
+        <username>{{your-github-username}}</username>
+        <password>{{your-github-personal-access_token}}</password>
+    </server>
+</servers>
+```
+
+Then, add the following repository block to `pom.xml` in the created Pulumi project (`pulumi-kotlin-demo`):
+
+```xml
+
+<repository>
+    <id>github-pulumi-kotlin</id>
+    <name>GitHub VirtuslabRnD/pulumi-kotlin Apache Maven Packages</name>
+    <url>https://maven.pkg.github.com/virtuslabrnd/pulumi-kotlin</url>
+</repository>
+```
+
+#### 4. Add new or replace existing dependencies.
+
+> ℹ️ See [the section below](#supported-providers) section for the full list of supported providers along with docs.
+
+Copy the `dependency` block from [the table below (Supported providers)](#supported-providers) or from 
+[GitHub Packages page for `pulumi-kotlin`](https://github.com/orgs/VirtuslabRnD/packages?repo_name=pulumi-kotlin). 
+Then, paste it to your `pom.xml`. For example:
+
+ ```xml
+
+<dependency>
+    <groupId>com.virtuslab</groupId>
+    <artifactId>pulumi-google-native-kotlin</artifactId>
+    <version>0.27.0.0</version>
+</dependency>
+ ```
+
+Only Kotlin-specific dependency is needed, so you can replace:
+
+  ```xml
+
+<dependency>
+    <groupId>com.pulumi</groupId>
+    <artifactId>aws</artifactId>
+    <version>5.16.2</version>
+</dependency>
+ ```
+
+with:
+
+ ```xml
+
+<dependency>
+    <groupId>com.virtuslab</groupId>
+    <artifactId>pulumi-aws-kotlin</artifactId>
+    <version>5.16.2.0</version>
+</dependency>
+ ```
+
+#### 5. Write the code.
+
+For inspiration, see the 
+["Provisioning a virtual machine on Google Cloud Platform"](#provisioning-a-virtual-machine-on-google-cloud-platform)
+example above (it's from Pulumi Java docs for Google Native provider, rewritten to Pulumi Kotlin).
+
+Using IntelliJ is heavily recommended as it can guide you through type-safe Kotlin DSLs.
+
+Additionally, there are two sources of documentation:
+
+- Pulumi official docs (example for [`google-native`](https://www.pulumi.com/registry/packages/google-native))
+- Kotlin API docs (example for [`google-native`](https://storage.googleapis.com/pulumi-kotlin-docs/google-native/0.27.0.0/index.html))
+
+#### 6. Deploy to the cloud.
+
+Run `pulumi up`. This provisions all cloud resources declared in your code.
+
+## Supported providers
```

Comment:
It might be worth noting somewhere that the code snippets in the docs only include Java examples for now?